modules/distributor/distributor.go

```diff
@@ -353,12 +360,12 @@ func (d *Distributor) PushTraces(ctx context.Context, traces ptrace.Traces) (*te
 	return nil, nil // PushRequest is ignored, so no reason to create one
 }
 
-func (d *Distributor) sendToIngestersViaBytes(ctx context.Context, userID string, traces []*rebatchedTrace, keys []uint32) error {
-	// Marshal to bytes once
+func (d *Distributor) sendToIngestersViaBytes(ctx context.Context, userID string, traces []*rebatchedTrace, keys []uint32, totalSpanCount int, tracesByID map[uint32]*rebatchedTrace) error {
 	marshalledTraces := make([][]byte, len(traces))
 	for i, t := range traces {
 		b, err := d.traceEncoder.PrepareForWrite(t.trace, t.start, t.end)
 		if err != nil {
+			overrides.RecordDiscardedSpans(totalSpanCount, reasonInternalError, userID)
```

Comment:
i think we should leave this outside `sendToIngesterViaBytes`. Basically either:

- `sendToIngesterViaBytes` returns an error meaning something unexpected happened and we can record drops due to internal errors.
- `sendToIngesterViaBytes` returns nil and internally it will record drops due to policy reasons.

i'm wondering if we want to completely stop recording internal error in metrics, and just rely on the error return. that would be for a future PR.

Reply:
Have we ever used this for any debugging purposes?

Reply:
Not that I'm aware of, but that metric is exposed to our cloud users so we would want to consider the impact before changing it.